### DISCUSSION TITLE
[Feature] Output Skipped Packages when `--skipPackageJson` Provided

### DISCUSSION THREAD

**Discussion by Char2sGu** [nrwl/nx#22935] [Feature Requests]
Developers like myself prefer to always enable `--skipPackageJson` to have full control over the packages installed, since Nx plugins are too opinionated and installs too many irrelevant dependencies.

However, currently I have to first execute the generator without `--skipPackageJson` to see what dependencies are needed, revert all changes, and then execute the generator again with `--skipPackageJson` to manage the dependencies myself, which is not qutie convenient.

As a solution, I propose that all `--skipPackageJson`, or `--skipPackageJson` for generators solely from `@nx/angular` if that's not possible, should output a list of skipped dependencies and the desired versions when `--skipPackageJson` is provided:

```sh
> nx g @nx/angular:lib my-lib --skipPackageJson
< xxxxx....
< Skipped Dependenceis:
< ng-packagr@x.x.x
< post-css@x.x.x
```

Alternatively, add support for a new option such as `--showPackageJsonChanges`. When supplied, Nx will output the intended changes to package.json, no matter whether `--skipPackageJson` is specified.

**Comment by jaysoo** [comment]
What `package.json` updates do you think are unnecessary that you often opt out of? You could run without `--skipPackageJson`, and git checkout the `package.json` file (or cherry pick what you want to include) after Nx updates it.

**Reply by Char2sGu** [reply]
> You could run without --skipPackageJson, and git checkout the package.json file (or cherry pick what you want to include) after Nx updates it.

I am doing that as a workaround, and I don't think this is the best way to fine-tune dependenceis. If `--skipPackageJson` isn't meant for this, what are its use cases? In my opinion, printing the dependency changes makes things clear in any code generation operations.

**Reply by Char2sGu** [reply]
> What package.json updates do you think are unnecessary that you often opt out of?

Nx tends to add all the peer dependencies of Nx plugins as workspace dependencies, which messes up the workspace package.json and makes it hard to determine which dependencies are extraneous. Our convention is to add only the dependencies that are directly referenced in our code to the workspace package.json.

**Reply by Char2sGu** [reply]
@jaysoo Another proposal is to add a `--skipInstall` option. When enabled, Nx modifieds package.json but does not automatically update the dependencies, which allows the developer to revise the package.json before installing.
